src/tests/catwalk_tests/test_evaluation.py

```diff
@@ -1,22 +1,243 @@
+            and model_id = '{model_id}'
+        """,
+        "hash": "klmno",
+    },
+    {
+        "name": "odds",
+        "query": """\
+            select entity_id, as_of_date
+            from {results_schema}.predictions
+            where entity_id % 2 = 1
+            and as_of_date in (SELECT(UNNEST(ARRAY{as_of_dates}::timestamp[])))
+            and model_id = {model_id}
+        """,
+        "hash": "fghij",
+    },
+    {
+        "name": "empty",
+        "query": """\
+            select entity_id, as_of_date
+            from {results_schema}.predictions
+            where entity_id = -1
+            and as_of_date in (SELECT(UNNEST(ARRAY{as_of_dates}::timestamp[])))
+            and model_id = {model_id}
+        """,
+        "hash": "abcde",
+    },
+]
+
+
+def setup_results_schema(db_engine, num_entities, labels):
+    # set up data, randomly generated by the factories but conforming
+    # generally to what we expect model_metadata and results schemas data
+    # to look like
+    model = ModelFactory()
+
+    class ImmediateTestPredictionFactory(TestPredictionFactory):
+        as_of_date = factory.LazyAttribute(
+            lambda o: o.model_rel.train_end_time
+        )
+    class ImmediateTrainPredictionFactory(TrainPredictionFactory):
+        as_of_date = factory.LazyAttribute(
+            lambda o: o.model_rel.train_end_time
+        )
+    
+    for entity in range(0, num_entities):
+        ImmediateTestPredictionFactory(
+            model_rel=model,
+            entity_id=entity,
+            label_value=labels[entity],
+        )
+        ImmediateTrainPredictionFactory(
+            model_rel=model,
+            entity_id=entity,
+            label_value=labels[entity],
+        )
+
+    session.commit()
+        
+    scores = []
+    labels = []
+    for row in db_engine.execute(
+        """\
+        select label_value, score
+        from test_results.predictions
+        order by entity_id
+        """
+    ):
+        labels.append(row[0])
+        scores.append(float(row[1]))
+
+    model_metadata = [
+        row 
+        for row in db_engine.execute(
+            """\
+            select model_id, train_matrix_uuid, train_end_time
+            from model_metadata.models
+            """
+        )
+    ][0]
+
+    return scores, model_metadata[0], model_metadata[1], model_metadata[2]
+
+
+def fake_matrix_from_db(db_engine, matrix_type, model_id, num_entities, as_of_date):
+        fake_matrix_store = MockMatrixStore(
+            matrix_type,
+            model_id,
+            num_entities,
+            db_engine,
+            init_labels=pandas.read_sql(
+                "select entity_id, as_of_date, label_value from train_results.predictions",
+                db_engine
+            ).set_index(["entity_id", "as_of_date"]).label_value,
+            init_as_of_dates=[as_of_date]
+        )
+        return fake_matrix_store
```

Comment:
very much up to the author, but there's no need to always set a local variable -- this can just return --

```
return MockMatrixStore(
    ...
)
```

Reply:
Implemented in latest push.